src/excmds.ts

```diff
@@ -3072,7 +3072,38 @@ export async function bmark(url?: string, ...titlearr: string[]) {
     dupbmarks.map(bookmark => browser.bookmarks.remove(bookmark.id))
     if (dupbmarks.length != 0) return
     const path = title.substring(0, title.lastIndexOf("/") + 1)
-    // TODO: if title is blank, get it from the page.
+
+
+    //this whole block grabs the current tab's title
+    //based on an async call since document.title won't have any meaning here
+    //
+    var currTitle;
+    function getCurrTabTitle(tabs) {
+        //function callback for promise for getting title of current tab
+        //requires [tabs] permission which tridactyl should have grabbed anyway
+        if (tabs[0]) {
+            var currentTab = tabs[0];
+            currTitle = currentTab.title
+        }
+    }
+    var titleSourceTab = browser.tabs.query({active: true, currentWindow: true});
+    //TODO: define and implement sane behaviour for if the title is blank, but the URL
+    //      that is being bookmarked is not actually the current tab. just using 
```

Comment:
if `url` is defined, don't bother trying to find a title, just use the URL. Similarly, if the tab title is undefined, use the URL.

We could do more complicated things with requests but I'm not too bothered. I suspect most people only ever bookmark the current page they are on, or add a bookmark in their RC and specify a title.